aea/cli/add.py

```diff
@@ -181,58 +155,124 @@ def _add_item(click_context, item_type, item_public_id) -> None:
             item_type, public_id=item_public_id, cwd=ctx.cwd, dest=dest_path
         )
 
+    item_config = _load_item_config(item_type, package_path)
+
+    if not _is_fingerprint_correct(package_path, item_config):
+        raise click.ClickException("Failed to add an item with incorrect fingerprint.")
+
+    _add_item_deps(click_context, item_type, item_config)
+    _register_item(ctx, item_type, item_public_id)
+
+
+def _is_item_present(ctx: Context, item_type: str, item_public_id: PublicId) -> bool:
```

Comment:
Seems like the `Configuration` should expose a method to check if any item of item type is present in it. Then this method is no longer needed.

Reply:
Requires Configuration upgrade. Currently AgentConfig (I suppose this method should belong to this class) doesn't have a property for cwd, well it seems it has `directory` property but in it's code I see that is never assigned and always None. So currently we can't check a package presence in agent folder.
I propose to leave it for now here and remove after AgentConfig upgrade so we can use a cwd from it's props (worth to create a ticket for that).
Correct me if I'm wrong.